src/game/game.cpp

```diff
@@ -528,12 +528,32 @@ namespace game
 
     bool inzoom()
     {
-        if(lastzoom && (zooming || lastmillis-lastzoom <= W(focus->weapselect, cookzoom)))
-            return true;
-        return false;
+        return lastzoom && (zooming || lastmillis - lastzoom <= W(focus->weapselect, cookzoom));
     }
     ICOMMAND(0, iszooming, "", (), intret(inzoom() ? 1 : 0));
 
+    // Get the progress of the zoom in the interval [0, 1], where 0 is no zoom and 1 is full zoom.
+    float zoomratio(gameent *d)
+    {
+        int frame = lastmillis - lastzoom;
+
+        int cookzoom = W(d->weapselect, cookzoom);
+
+        float ratio = 1.f;
+        if (cookzoom > 0 && frame <= cookzoom) {
+            ratio = frame / float(W(d->weapselect, cookzoom));
+        }
+
+        ratio = clamp(ratio, 0.f, 1.f);
+
+        if(zooming) {
```

Comment:
Minor style issue: we're putting braces below the statement, at least for now until we have proper code style guidelines. This follows the existing "style".

Reply:
I'd like to see some doc strings in that function. You should describe what `zooming` does for example (indicates whether you're zooming in or out, basically, not whether you're zooming in either direction; maybe we should rename it).